test/unit/mem/view/src/Accessor.cpp

```diff
@@ -0,0 +1,373 @@
+        {
+            data[1] = 1.0f;
+            data(2) = 2.0f;
+            data[alpaka::Vec<alpaka::DimInt<1>, TIdx>{TIdx{3}}] = 3.0f;
+        }
+    };
+
+    struct ReadKernelTemplate
+    {
+        template<typename TAcc, typename TAccessor>
+        ALPAKA_FN_ACC void operator()(TAcc const&, TAccessor data) const
+        {
+            float const v1 = data[1];
+            float const v2 = data(2);
+            float const v3 = data[alpaka::Vec<alpaka::DimInt<1>, alpaka::Idx<TAcc>>{alpaka::Idx<TAcc>{3}}];
+            (void) v1;
+            (void) v2;
+            (void) v3;
+        }
+    };
+
+    struct ReadKernelExplicit
+    {
+        template<typename TAcc, typename TMemoryHandle, typename TIdx>
+        ALPAKA_FN_ACC void operator()(
+            TAcc const&,
+            alpaka::Accessor<TMemoryHandle, float, TIdx, 1, alpaka::ReadAccess> const data) const
+        {
+            float const v1 = data[1];
+            float const v2 = data(2);
+            float const v3 = data[alpaka::Vec<alpaka::DimInt<1>, TIdx>{TIdx{3}}];
+            (void) v1;
+            (void) v2;
+            (void) v3;
+        }
+    };
+
+    struct ReadWriteKernelExplicit
+    {
+        template<typename TAcc, typename TMemoryHandle, typename TIdx>
+        ALPAKA_FN_ACC void operator()(
+            TAcc const&,
+            alpaka::Accessor<TMemoryHandle, float, TIdx, 1, alpaka::ReadWriteAccess> const data) const
+        {
+            float const v1 = data[1];
+            float const v2 = data(2);
+            float const v3 = data[alpaka::Vec<alpaka::DimInt<1>, TIdx>{TIdx{3}}];
+            (void) v1;
+            (void) v2;
+            (void) v3;
+
+            data[1] = 1.0f;
+            data(2) = 2.0f;
+            data[alpaka::Vec<alpaka::DimInt<1>, TIdx>{TIdx{3}}] = 3.0f;
+        }
+    };
+} // namespace
+
+TEST_CASE("readWrite", "[accessor]")
+{
+    using Dim = alpaka::DimInt<1>;
+    using Size = std::size_t;
+    using Acc = alpaka::ExampleDefaultAcc<Dim, Size>;
+    using DevAcc = alpaka::Dev<Acc>;
+    using Queue = alpaka::Queue<DevAcc, alpaka::Blocking>;
+
+    auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+    auto queue = Queue{devAcc};
+    auto buffer = alpaka::allocBuf<float, Size>(devAcc, Size{N});
+    auto const workdiv = alpaka::WorkDivMembers<Dim, Size>{
+        alpaka::Vec<Dim, Size>{Size{1}},
+        alpaka::Vec<Dim, Size>{Size{1}},
+        alpaka::Vec<Dim, Size>{Size{1}}};
+
+    alpaka::exec<Acc>(queue, workdiv, WriteKernelTemplate{}, alpaka::writeAccess(buffer));
+    alpaka::exec<Acc>(queue, workdiv, WriteKernelExplicit{}, alpaka::writeAccess(buffer));
+    alpaka::exec<Acc>(queue, workdiv, ReadKernelTemplate{}, alpaka::readAccess(buffer));
+    alpaka::exec<Acc>(queue, workdiv, ReadKernelExplicit{}, alpaka::readAccess(buffer));
+    alpaka::exec<Acc>(queue, workdiv, ReadWriteKernelExplicit{}, alpaka::access(buffer));
```

Comment:
Can or should I be able to call `alpaka::exec<Acc>(queue, workdiv, ReadKernelExplicit{}, alpaka::access(buffer));` or `alpaka::exec<Acc>(queue, workdiv, WriteKernelExplicit{}, alpaka::access(buffer));`